java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -236,8 +235,6 @@ public ArrowBuf buffer(final int initialRequestSize) {
   }
 
   private ArrowBuf createEmpty() {
-    assertOpen();
```

Comment:
this did not seem necessary either, because the an allocator will always hold an instance of an empty buffer.  Does this seem ok?